Geometry/HGCalGeometry/interface/HGCalMouseBite.h

```diff
@@ -0,0 +1,33 @@
+#ifndef GeometryHGCalGeometry_HGCalMouseBite_h
+#define GeometryHGCalGeometry_HGCalMouseBite_h
+
+#include "DataFormats/ForwardDetId/interface/HFNoseDetId.h"
+#include "DataFormats/ForwardDetId/interface/HGCSiliconDetId.h"
+#include "Geometry/HGCalCommonData/interface/HGCalDDDConstants.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#include <vector>
+
+//#define EDM_ML_DEBUG
+
+class HGCalMouseBite {
+public:
+  HGCalMouseBite(const HGCalDDDConstants& hgc, bool waferRotate = false);
+  template <class T>
+  bool exclude(const T& id) {
+    int iuv = (100 * id.cellU() + id.cellV());
+    bool check =
+        ((id.type() == 0) ? (std::find(rejectFine_.begin(), rejectFine_.end(), iuv) != rejectFine_.end())
```

Comment:
since the `rejectFine_` and `rejectCoarse_` vectors are only filled once in the constructor of this class, it would be much more efficient to sort them once they are filled, and then perform this check using `std::binary_search` or a similar log(n) algorithm.

Reply:
They are any way sorted the way it is constructed. So I am using binary_search here